server/src/test/java/org/opensearch/index/store/RemoteSegmentStoreDirectoryTests.java

```diff
@@ -364,6 +369,96 @@ public void testCopyFrom() throws IOException {
         storeDirectory.close();
     }
 
+    public void testCopyFilesFromMultipart() throws Exception {
+        String filename = "_100.si";
+        populateMetadata();
+        remoteSegmentStoreDirectory.init();
+
+        Directory storeDirectory = LuceneTestCase.newDirectory();
+        IndexOutput indexOutput = storeDirectory.createOutput(filename, IOContext.DEFAULT);
+        indexOutput.writeString("Hello World!");
+        CodecUtil.writeFooter(indexOutput);
+        indexOutput.close();
+        storeDirectory.sync(List.of(filename));
+
+        assertFalse(remoteSegmentStoreDirectory.getSegmentsUploadedToRemoteStore().containsKey(filename));
+
+        BlobContainer blobContainer = mock(BlobContainer.class);
+        when(remoteDataDirectory.getBlobContainer()).thenReturn(blobContainer);
+        when(blobContainer.isMultiStreamUploadSupported()).thenReturn(true);
+        CompletableFuture<UploadResponse> uploadResponseCompletableFuture = new CompletableFuture<>();
+        uploadResponseCompletableFuture.complete(new UploadResponse(true));
+        when(blobContainer.writeBlobByStreams(any(WriteContext.class))).thenReturn(uploadResponseCompletableFuture);
+
+        remoteSegmentStoreDirectory.copyFilesFrom(storeDirectory, List.of(filename), IOContext.DEFAULT);
+
+        assertTrue(remoteSegmentStoreDirectory.getSegmentsUploadedToRemoteStore().containsKey(filename));
```

Comment:
can we assert the `segmentsUploadedToRemoteStore` updation as well ? we can hep a test specific function for that .

Reply:
https://github.com/opensearch-project/OpenSearch/issues/8080